src/main/java/io/xpring/ilp/IlpClient.java

```diff
@@ -0,0 +1,101 @@
+import org.interledger.spsp.server.grpc.SendPaymentResponse;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.xpring.xrpl.XpringException;
+
+import java.math.BigInteger;
+import java.util.Optional;
+
+/**
+ * A client that can create accounts, get accounts, get balances, and send ILP payments on a connector.
+ */
+public class IlpClient {
+    private IlpClientDecorator decoratedClient;
+
+    /**
+     * Initialize a new client with the given options.
+     *
+     */
+    public IlpClient() {
+        this.decoratedClient = new DefaultIlpClient();
+    }
+
+    /**
+     * Constructor meant only for integration testing, so that we can point the client at a docker container host
+     * @param grpcUrl : Url of the Hermes host within a docker container
+     *
+     */
+    @VisibleForTesting
+    protected IlpClient(String grpcUrl) {
+        this.decoratedClient = new DefaultIlpClient(grpcUrl);
+    }
+
+    /**
+     * Create an account on the connector.  Because no settings or auth were passed in, an account
+     * with default settings, a generated account ID, and a generated SIMPLE auth token will be generated for you
+     *
+     * @return A {@link CreateAccountResponse} containing the account settings that were created on the connector
+     * @throws XpringException If account creation failed.
+     */
+    public CreateAccountResponse createAccount() throws XpringException {
+        return decoratedClient.createAccount();
+    }
+
+    /**
+     * Create an account on the connector.
+     *
+     *
+     * @param createAccountRequest: A request object with specified account details. Note that {@link CreateAccountRequest#assetScale}
+     *                           and {@link CreateAccountRequest#assetCode} MUST be specified, as guaranteed by {@link CreateAccountRequest#builder(String, Integer)}
+     * @return A {@link CreateAccountResponse} containing the account settings that were created on the connector
+     * @throws XpringException If the given inputs were invalid or account creation failed.
+     */
+    public CreateAccountResponse createAccount(CreateAccountRequest createAccountRequest, Optional<String> bearerToken) throws XpringException {
+        return decoratedClient.createAccount(createAccountRequest, bearerToken);
+    }
+
+    /**
+     * Gets account details for the given account ID
+     *
+     * @param accountId
+     * @param bearerToken
+     * @return
+     * @throws XpringException
+     */
+    public GetAccountResponse getAccount(String accountId, String bearerToken) throws XpringException {
+        return decoratedClient.getAccount(accountId, bearerToken);
+    }
+
+    /**
+     * Get the balance of the specified account on the connector.
+     *
+     * @param accountId The account ID to get the balance for.
+     * @param bearerToken Authentication bearer token. TODO: Probably change from string to some wrapped JWT class
```

Comment:
Assign an owner here

Reply:
Will do this for the other TODOs, but I'm just going to get rid of this one.  I think we decided that there's no use in wrapping the auth token in some sort of primitive